### ISSUE TITLE
Can we combine this profitably with Schedule-Free?

### ISSUE THREAD

**Issue by JDvorak** [konstmish/prodigy#17]
https://github.com/facebookresearch/schedule_free
 
 This looks compatible with the prodigy approach, but not entirely sure.

**Comment by konstmish** [comment]
Working on that!

**Comment by dxqbYD** [comment]
I've created an experimental implementation of schedule-free Prodigy. First tests are going well:
![grafik](https://github.com/user-attachments/assets/562d4826-7501-4906-8788-cc2cf9f4cca2)

**Comment by dxqbYD** [comment]
https://github.com/konstmish/prodigy/pull/26

**Comment by JaheimLee** [comment]
Hi, any progress here? @konstmish

**Comment by konstmish** [comment]
Yes, I think we'll have the Schedule-Free version in a couple of weeks in the repo. At the moment, the priority is to change the implementation of the main optimizer and test it as outlined [here](https://github.com/konstmish/prodigy/pull/28#issuecomment-2578304552) (tasks 2 and 3; Schedule-Free is number 5).
